crates/merkle-tree/src/tree.rs

```diff
@@ -589,12 +603,12 @@ impl<H: FeltHash, const HEIGHT: usize> MerkleTree<H, HEIGHT> {
                         height += 1;
 
                         let left = storage
-                            .hash(left)
+                            .hash(left.get())
```

Comment:
Can't we change the `pathfinder_merkle_tree::storage::Storage` trait to use `TrieStorageIndex` as arguments to `get()` and `hash()`? That seems to be another place where we're still having trie storage indices passed as `u64`. That would also allow us removing many of these conversions in this file.